packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -129,15 +137,24 @@ library LibProving {
                 ] = fcId;
             }
         } else if (evidence.prover == address(0)) {
+            // This is the branch the oracle prover is trying to overwrite
             fc = blk.forkChoices[fcId];
-
             if (
                 fc.blockHash == evidence.blockHash &&
                 fc.signalRoot == evidence.signalRoot &&
                 fc.gasUsed == evidence.gasUsed
             ) revert L1_SAME_PROOF();
         } else {
-            revert L1_ALREADY_PROVEN();
+            // This is the branch provers trying to overwrite
+            fc = blk.forkChoices[fcId];
+            if (fc.prover != address(0) && fc.prover != address(1))
+                revert L1_ALREADY_PROVEN();
```

Comment:
lets say `realProofSkipSize` is `10`, and right now we have a new block whose id is `10`. Then we submit an oracle proof to the protocol, after that, we need to submit a real proof for that block, but the real proof submission will run into this `else` logic and get a `L1_ALREADY_PROVEN` custom error?

Reply:
With the current code,  the error will not occur. As `fc.prover == address(0)`.